src/controllers/signinController.ts

```diff
@@ -10,7 +10,12 @@ export const signinController = async (req: Request, res: Response) => {
 
     if (user) {
       const token = generateToken(user);
-      res.json({ success: true, token });
+      if (token) {
+        res.status(200).json({ success: true, token });
+      } else {
+        throw new Error('Login failed');
```

Comment:
should these respond by adding a status code and message to the response object instead? Not sure what Express' error handling looks like for this

Reply:
Good question @briri . When you throw an error here, it will be caught by the "catch" in the "try-catch" that it's in. This is what will be logged: `"{"level":50,"time":1720547549009,"pid":42151,"hostname":"CDL-JSHIN-9M","err":{"type":"Error","message":"Login failed","stack":"Error: Login failed\n    at signinController (/Users/jshin/dev/dmsp_backend_prototype/src/controllers/signinController.ts:16:15)"},"msg":"Signin error"}"` and it will return a 500 error with the message "Internal Server Error".